### PR TITLE
try to get test to error in order to investigate #589

### PR DESCRIPTION
**Pull request by jgabry**
This PR should _not_ be merged, just using it to trigger the CI.  I'm trying to investigate the question in #589 as to why a test that should have failed in all runs only failed in test coverage on linux. This attempts to recreate the problem.

### CODE REVIEW

**Comment by jgabry** [comment]
Ok @rok-cesnovar this is pretty strange. I just intentionally inserted a function that doesn't exist into a test and yet all runs except linux test coverage pass. This replicates the issue described in issue #589 that happened in PR #587

File: tests/testthat/test-install.R [hunk 1]
```diff
@@ -26,6 +26,9 @@ test_that("install_cmdstan() successfully installs cmdstan", {
 
 test_that("install_cmdstan() errors if installation already exists", {
   skip_if_offline()
+  if (not_on_cran()) {
```
Comment (jgabry) [inline tests/testthat/test-install.R:29]: this function doesn't exist anymore and so should cause an error!

Comment (jgabry) [reply tests/testthat/test-install.R:29]: the only thing I can think of is maybe `skip_if_offline()` is being triggered? but that doesn't really make any sense if cmdstan was successfully installed from the internet in each of these CI runs. I will try removing `skip_if_offline()` just to see what happens.

**Comment by jgabry** [comment]
Ok very strange. It looks like the test was indeed being skipped because of `skip_if_offline`: once I commented out `skip_if_offline` the test started failing on other runs (as of this comment it's now failing on mac, which it wasn't before. We'll see about the others when they finish). What I don't understand is 

* why does it think it's offline? `skip_if_offline` checks `has_internet <- !is.null(curl::nslookup(host, error = FALSE))` where `host="r-project.org"`. 
* why did this previously only fail on the linux test coverage run?

I think it's probably safe to remove all uses of `skip_if_offline`. At least I'll do that on this branch so we can see if the `install_cmdstan` tests pass (`skip_if_offline` was only being used in those tests).

**Comment by jgabry** [comment]
Without `skip_if_offline` we now have tests failing just on windows in r cmd check but apparently not failing on windows in the separate test coverage run and not failing with the other OSes

**Comment by jgabry** [comment]
@rok-cesnovar No rush, but at some point can you take a look at the windows runs that are now failing. My lack of windows knowledge is preventing me from figuring out what's going on here.

**Comment by jgabry** [comment]
Thanks! I think after my debugging it's pretty clear we need to remove all uses of `skip_if_offline()` (still a mystery as to why that wasn't working properly), but I couldn't quite figure out why that was now causing the windows failures. Thanks for looking into it.

**Comment by jgabry** [comment]
Just noticed this from one of the earlier failed mac runs: 

https://github.com/stan-dev/cmdstanr/runs/4157825884?check_suite_focus=true#step:17:189

It says 

```
══ Skipped tests ═══════════════════════════════════════════════════════════════
• !mpi_toolchain_present() is TRUE (1)
• curl cannot be loaded (5)
```

so I guess it was failing to load curl and that's why `skip_if_offline()` was triggered. Still not sure why it was failing to load curl. Anyway, now that we're removing `skip_if_offline()` this is kind of irrelevant.

**Comment by rok-cesnovar** [comment]
So the issue was that older CmdStan versions (pre-2.26) had an issue with one of the dependency libraries (Boost Program Options) and the GHA setup is done in such a way that building failed. That issue went away with 2.26 so I just bumped the versions used in the tests, so should be good now.

**Comment by jgabry** [comment]
Ok great, thanks for sorting that out. I guess we should we should merge this then. I'll convert it from a draft and then merge.

**Pull request merged**